pkg/cmd/pulumi/deployment.go

```diff
@@ -179,90 +244,56 @@ func newDeploymentSettingsInitCmd() *cobra.Command {
 		Short:      "Initialize the stack's deployment.yaml file",
 		Long:       "",
 		Run: cmdutil.RunFunc(func(cmd *cobra.Command, args []string) error {
-			ctx := cmd.Context()
-			interactive := cmdutil.Interactive()
-
-			displayOpts := display.Options{
-				Color:         cmdutil.GetGlobalColorization(),
-				IsInteractive: interactive,
-			}
-
-			project, _, err := readProject()
-			if err != nil && !errors.Is(err, workspace.ErrProjectNotFound) {
-				return err
-			}
-
-			currentBe, err := currentBackend(ctx, project, displayOpts)
+			ctx, display, s, _, be, err := initializeDeploymentSettingsCmd(cmd, stack)
 			if err != nil {
 				return err
 			}
 
-			if !currentBe.SupportsDeployments() {
-				return fmt.Errorf("backends of this type %q do not support deployments",
-					currentBe.Name())
+			newStackDeployment := &workspace.ProjectStackDeployment{
+				DeploymentSettings: apitype.DeploymentSettings{},
 			}
 
-			s, err := requireStack(ctx, stack, stackOfferNew|stackSetCurrent, displayOpts)
+			err = configureGit(ctx, display, be, s, newStackDeployment, gitSSHPrivateKeyPath)
 			if err != nil {
 				return err
 			}
 
-			wd, err := os.Getwd()
+			err = configureImageRepository(ctx, display, be, s, newStackDeployment)
 			if err != nil {
 				return err
 			}
-			repo, err := git.PlainOpen(wd)
+
+			err = configureAdvancedSettings(display, newStackDeployment)
 			if err != nil {
 				return err
 			}
 
-			newStackDeployment := &workspace.ProjectStackDeployment{
-				DeploymentSettings: apitype.DeploymentSettings{
-					SourceContext: &apitype.SourceContext{
-						Git: &apitype.SourceContextGit{
-							// Setting this by default to be included in the deployment file
-							// so it is easier to manually change by users. This will be completed
-							// by the user when this gets converted to a wizard.
-							RepoDir: ".",
-						},
-					},
-					Operation: &apitype.OperationContext{
-						Options: &apitype.OperationContextOptions{},
-					},
+			var option string
+			if err := survey.AskOne(&survey.Select{
```

Comment:
I know we use survey across the board but it's recently been archived. Wonder if there's any movement around migrating to bubbletea as recommended in survey's [repo](https://github.com/AlecAivazis/survey#survey).

Reply:
There is no 1:1 replacement and I would prefer platform to make the call on how to replace these. I believe https://github.com/charmbracelet/huh would be a better replacement (it is a very good abstraction over bubbletea with a nice API to use) but still there might be small changes in behaviour and style, so I think it should be done altogether to avoid having commands looking different than others